OasisDEX/oasis-borrow
# introduced constant multiple read only safe switch flag

# [Constant Multiple Readonly safety switch](https://app.shortcut.com/oazo-apps/story/5631/safety-switch)

NOTE to test readonly flag CM feature flag should also be on
  
## Changes 👷‍♀️
-introduced constantMultipleReadOnly flag
  
## How to test 🧪
  
- Both constant multiple and constant multiple read only off user shouldn't see any CM features|
![image](https://user-images.githubusercontent.com/6871923/185387540-65e02fe7-89ee-47e4-b20c-bc3b4a27111d.png)

- all CM and BS flags off user is informed that it's not possible to access optimization at the moment

![image](https://user-images.githubusercontent.com/6871923/185387947-22398b47-43cf-4003-b034-122f40052759.png)

- BasicBS on but Constant Multiple Read Only Enabled, user has no trigger - constant multiple remains hidden

- BasicBS is on , CM is in readonly but user has trigger - message shall be displayed, user allowed to remove trigger
![image](https://user-images.githubusercontent.com/6871923/185601240-fd210116-bf09-4619-9e05-5b8fa96433ff.png)

![image](https://user-images.githubusercontent.com/6871923/185601315-53bd42c2-c294-4468-8ca2-c3a23cd8480a.png)

https://goerli.etherscan.io/tx/0xe842c7c2ae1c9dc680cec02a1fca99700ec42b78117b733befa13a2aacc4a17f

BS is on but CM is in RO, user has no trigger, CM banner is hidden
![image](https://user-images.githubusercontent.com/6871923/185895935-e29544f0-bf5e-4b4e-9885-e668f8517c50.png)


BS, enabled, CM in RO, trigger exist - user allowed to remove existing trigger
![image](https://user-images.githubusercontent.com/6871923/185898012-5986dbe2-227f-4854-bc9c-fd578703ad35.png)

![image](https://user-images.githubusercontent.com/6871923/185898093-dc0ecba3-2ebb-45d7-b64b-5b42bef33856.png)
